DxDiagram new API (#1112)
DevExpress/devextreme-documentation: api-reference/10 UI Widgets/dxDiagram/1 Configuration/edges/customDataExpr.md

---
---
##### shortDescription
Specifies the name of a data source field or an expression that provides an edge's custom data.

##### param(data): Object
The current edge's data object.

---

> This option links custom data contained in a data source to a diagram edge. The linked data is copied and pasted with the edge. Changes in the data are reflected in a diagram history and can be undone and redone by the UI.

Reviewer: "The linked data is copied and pasted with the edge."

Can you clarify what you mean here?

Author: A diagram item (node or edge) can contain data that are not visible in the diagram, e.g., detailed information about a person.  If required, a user can access this data directly (without this option) from a data source, but in this case the widget does not copy-pasted the custom data with the item. If user specifies this option, the custom data is linked to an item and a copy-pasted item contain all the custom information.

Reviewer: Thank you. Could we say:

```suggestion
This option links custom data from a data source to the diagram edge. The edge contains the linked data copied from the data source. Changes in the data are reflected in the diagram history. You can use the UI to undo and redo these changes.
```
